app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -153,6 +184,68 @@ class UserViewModel(
     }
   }
 
+  /**
+   * Upload the picture to the cloud storage update the user object in the database with the cloud
+   * path created. update the user object in the viewmodel with the url of the image and the cloud
+   * path
+   */
+  private fun uploadProfilePicture(context: Context) {
+    if (imageRepository == null) {
+      Log.e("UserViewModel", "ImageRepository is null")
+      return
+    }
+    val user = currentUser.value ?: return
```

Comment:
This function uploadPicture is never supposed to be called when a user is not set.

I can only see this happens in test if we made a mistake writing the code.

I could go harder and instead of just returning throwing a fatal exception what do you think ?

Reply:
Well if you are sure that this function should never be called when currentUser is null then you can. However i was thinking more of a Log which seems less "fatal" xD.

Reply:
Solved, added Log.e() + return if invalid arguments